src/screens/select.tsx

```diff
@@ -108,42 +98,61 @@ const SelectScreen: React.FC<SelectScreenProps> = ({ navigation }) => {
         setStatusMessage(JSON.stringify(error));
       });
   };
-  const incAdult = () => {
-    if (adults < 6) {
-      setAdults(adults + 1);
-    }
-  };
-  const decAdult = () => {
-    if (adults > 0) {
-      setAdults(adults - 1);
-    }
-  };
-  const incChildren = () => {
-    if (children < 6) {
-      setChildren(children + 1);
-    }
-  };
-  const decChildren = () => {
-    if (children > 0) {
-      setChildren(children - 1);
+  const showOrHidePicker = (dateOrTime: boolean, show: boolean) => {
+    if (dateOrTime && show) {
+      setDatePickerVisibility(true);
+    } else if (dateOrTime && !show) {
+      setDatePickerVisibility(false);
+    } else if (!dateOrTime && show) {
+      setTimePickerVisibility(true);
+    } else if (!dateOrTime && !show) {
+      setTimePickerVisibility(false);
     }
   };
-  const changeDate = (date: moment.Moment) => {
-    setSelectedDate(JSON.stringify(date.toDate()));
-  };
-  const showDatePicker = () => {
-    setDatePickerVisibility(true);
-  };
-
-  const hideDatePicker = () => {
-    setDatePickerVisibility(false);
+  const handleConfirmDate = (date: Date) => {
+    setSelectedDate(JSON.stringify(date));
+    showOrHidePicker(true, false);
   };
-
-  const handleConfirm = (date: Date) => {
+  const handleConfirmTime = (date: Date) => {
     setTime(JSON.stringify(date).substring(12, 20).replace(/:/g, '%3A'));
     console.log('A date has been picked: ', time);
-    hideDatePicker();
+    showOrHidePicker(false, false);
   };
+  useEffect(() => {
+    const fetchStations = () => {
+      console.log('hello');
+      fetch(stationsUrl, {
+        method: 'GET',
+        headers: {
+          'X-API-KEY': config.apiKey,
+          Accept: 'application/json',
+          'Content-Type': 'application/json',
+        },
+      })
+        .then((response) => response.json())
+        .then((response) => response as Stations)
+        .then((response) => {
+          if (response.error) {
```

Comment:
It'd be great to pull this out into a method and pass it into your then

Reply:
that's how we broke things earlier, we will attempt again in the remaining 20mins